fbw-a32nx/src/base/flybywire-aircraft-a320-neo/html_ui/Pages/VCockpit/Instruments/Airliners/FlyByWire_A320_Neo/FMC/A32NX_FMCMainDisplay.js

```diff
@@ -3121,162 +3126,243 @@ class FMCMainDisplay extends BaseAirliners {
+            return false;
         }
 
+        const accAlt = parseInt(match[1]);
+
+        const origin = this.flightPlanManager.getPersistentOrigin();
+        let elevation = origin.infos.elevation !== undefined ? origin.infos.elevation : 0;
         const minimumAltitude = elevation + 400;
 
-        if (thrRedAlt > 0) {
-            thrRedAlt = parseInt(thrRedAlt);
-            thrRedAlt = Math.round(thrRedAlt / 10) * 10;
-            currentThrRedAlt = thrRedAlt;
+        if (accAlt < minimumAltitude || accAlt > 45000) {
+            this.setScratchpadMessage(NXSystemMessages.entryOutOfRange);
+            return false;
+        }
+
+        this.flightPlanManager.getCurrentFlightPlan().engineOutAccelerationAltitudePilot = accAlt;
+
+        return true;
+    }
+
+    async trySetThrustReductionAccelerationAltitudeGoaround(s) {
+        if (this.flightPhaseManager.phase >= FmgcFlightPhases.GOAROUND || !this.flightPlanManager.getDestination()) {
+            this.setScratchpadMessage(NXSystemMessages.notAllowed);
+            return false;
         }
 
-        if (accAlt > 0) {
-            accAlt = parseInt(accAlt);
-            accAlt = Math.round(accAlt / 10) * 10;
-            currentAccAlt = accAlt;
+        if (s === FMCMainDisplay.clrValue) {
+            if (this.flightPlanManager.getCurrentFlightPlan().hasMissedThrustReductionAltitudeDefault && this.flightPlanManager.getCurrentFlightPlan().hasMissedAccelerationAltitudeDefault) {
+                this.flightPlanManager.getCurrentFlightPlan().missedThrustReductionAltitudePilot = undefined;
+                this.flightPlanManager.getCurrentFlightPlan().missedAccelerationAltitudePilot = undefined;
+                return true;
+            }
+            this.setScratchpadMessage(NXSystemMessages.notAllowed);
+            return false;
         }
 
-        if (thrRedAlt > 0 && (thrRedAlt < minimumAltitude || thrRedAlt > currentAccAlt)) {
-            this.setScratchpadMessage(NXSystemMessages.entryOutOfRange);
+        const match = s.match(/^([0-9]{4,5})?(\/([0-9]{4,5}))?$/);
+        if (match === null || (match[1] === undefined && match[3] === undefined)) {
+            this.setScratchpadMessage(NXSystemMessages.formatError);
             return false;
-        } else if (thrRedAlt > 0) {
-            updateThrRedAlt = true;
         }
 
-        if (accAlt > 0 && (accAlt < minimumAltitude || accAlt < currentThrRedAlt)) {
+        const thrRed = match[1] !== undefined ? FMCMainDisplay.round(parseInt(match[1]), 10) : undefined;
+        const accAlt = match[3] !== undefined ? FMCMainDisplay.round(parseInt(match[3]), 10) : undefined;
+
+        const destination = this.flightPlanManager.getDestination();
+        let elevation = destination.infos.elevation !== undefined ? destination.infos.elevation : 0;
+        const minimumAltitude = elevation + 400;
+
+        const plan = this.flightPlanManager.getCurrentFlightPlan();
```

Comment:
Maybe you can move this assignment to the start of the function and replace some of the `this.flightPlanManager.getCurrentFlightPlan()` calls.